File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/response/GetQuestionnaireResponse.java
package com.github.ai.leetcodequiz.api.response;

import com.github.ai.leetcodequiz.api.QuestionnaireItemDto;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record GetQuestionnaireResponse(
    @NotNull QuestionnaireItemDto questionnaire
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/response/GetQuestionnaireResponse.scala
package com.github.ai.leetcodequiz.api.response

import com.github.ai.leetcodequiz.api.{QuestionnaireItemDto, QuestionnairesItemDto}
import zio.json.{JsonDecoder, JsonEncoder}

case class GetQuestionnaireResponse(
  questionnaire: QuestionnaireItemDto
) derives JsonEncoder,
      JsonDecoder
